src/Algebra/Graph/AdjacencyMap/Extra.hs

```diff
@@ -43,3 +45,11 @@ shrinkSingle vert gr = AM.overlay (AM.removeVertex vert gr) inducedEdges
         | pre <- Set.toList (AM.preSet vert gr)
         , post <- Set.toList (AM.postSet vert gr)
         ]
+
+-- | Split graph into distinct sibling graphs via nodes that have no incoming edges.
+--
+-- @
+-- splitGraph 'edges' [(1, 2), (2, 3), (4, 5)] = ['edges' [(1, 2), (2, 3)], 'edge' 4 5]
+-- @
+splitGraph :: Ord a => AM.AdjacencyMap a -> [AM.AdjacencyMap a]
```

Comment:
- [ ] Unit test please

[nit optional] prefer naming to somewhat signify an approach - maybe `splitComponents` is maybe more apt?

Reply:
I don't think `splitComponents` offers any more info on the approach.  If you have a better name, I'm happy to change it, but I don't see how the word "components" offers any more clarity.